lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+const slider = document.getElementById("js-slider");
+
+const createAttributedElements = ({tag,attrObj,str}) =>{
+    const element = document.createElement(tag);
+    Object.keys(attrObj).forEach((attribute) => {
+        element.setAttribute(attribute, attrObj[attribute]);
+    });
+    if(str !== undefined) element.textContent = str;
+    return element;
+}
+
+const createErrorMessage = (error) => {
+    const errMsg = createAttributedElements({tag:"p",attrObj:{class:"error-msg"},str:error});
+    
+    slider.appendChild(errMsg);
+}
+
+const showLoading = () => {
+    const loadingPlace = createAttributedElements({tag:"div",attrObj:{id:"js-loading",class:"loading"}});
+    const loadingImage = createAttributedElements({tag:"img",attrObj:{src:"img/loading-circle.gif",alt:"ローディング"}});
+    
+    slider.appendChild(loadingPlace).appendChild(loadingImage);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
+}
+
+const fetchSliderData = (ms) => new Promise(resolve => setTimeout(() => resolve(fetchData("https://myjson.dit.upm.es/api/bins/2bzx")),ms));
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderElements(sliderData);
+        autoSwitchSlider(sliderData);
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderElements = (sliderData) => {
+    renderSliderItems(sliderData);
+    getPageNumber(sliderData);
+    renderButtons(sliderData);
+    renderDotPagination(sliderData);
+    isActiveFirstItem();
+}
+
+let autoPlay;
+const autoSwitchSlider = (sliderData) =>{
+    autoPlay = setInterval(() =>{
+        currentIndex ++;
+        if(currentIndex >= sliderData.length){
+            currentIndex = 0;
+        }
```

Comment:
Q.
Is this not good enough?
I would love to know!
```suggestion
        if(currentIndex === sliderData.length){
            currentIndex = 0;
        }
```

Reply:
The reason I used the equal operator is because I thought it was easy to understand.
However, even with the exact equality operator, the desired result is the same, 
which is sufficient, so I changed it.
https://github.com/nakagawamai/js-lesson/pull/20/commits/5e09e646635388e6b66ce53d1a4030c7f17f944c